### ISSUE TITLE
RefURIAttribute-Extension throws exception on ExternalInterfaceType

### ISSUE THREAD

**Issue by MenschMitHut** [AutomationML/AMLEngine2.1#5]
Hi,

the RefURIAttribute has a misleading behaviour. The ExternalInterfaceType is derived from the InterfaceClassType which implements the IObjectWithAttributes interface.
This allows me to use the RefURIAttribute-Extension. But it does not work.
You have to cast the ExternalInterface to an ObjectWithAMLAttributes to use its RefURIAttribute property and everything works as expected.
So there must be a bug in the extension.

```
var caexDocument = CAEXDocument.New_CAEXDocument();            

// add an InstanceHierarchy to the ROOT CAEXFile element
var myIH = caexDocument.CAEXFile.New_InstanceHierarchy("myIH");

// add an InternalElement
var ie = myIH.InternalElement.Append("ie");

ie.New_ExternalInterface("ColladaInterface").MakeCOLLADAInterface();

var colladaInterface = ie.ExternalInterface.First;

//works
var castedColladaInterface = (ObjectWithAMLAttributes)colladaInterface;
var nullAsExpected = castedColladaInterface.RefURIAttribute;

//throws exception
var refUriToDae = colladaInterface.RefURIAttribute(); 
```
```
System.InvalidCastException
  HResult=0x80004002
  Message=Das Objekt des Typs "Aml.Engine.CAEX.ExternalInterfaceType" kann nicht in Typ "Aml.Engine.AmlObjects.ObjectWithAMLAttributes" umgewandelt werden.
  Source=AML.Engine
  StackTrace:
   at Aml.Engine.AmlObjects.Extensions.AmlObjectsExtensions.RefURIAttribute(IObjectWithAttributes objectWithAttributes)
   at TestApp.Program.Main(String[] args) in ...\Visual Studio 2017\Projects\TestApp\Program.cs:line 35
```

**Comment by MenschMitHut** [comment]
Same issue with RefTypeAttribute-Extension

**Comment by EnderYemenicioglu** [comment]
Thank you for your bug report. We will fix this with the next version.

**Comment by jprinz** [comment]
Fixed in v1.0.17

**Comment by MenschMitHut** [comment]
Thanks
